docs/operational_guide/availability_consistency_durability.md

```diff
@@ -0,0 +1,71 @@
+# Tuning Availability, Consistency, and Durability
+
+## Overview
+
+Generally speaking M3DB is an [H.A](https://en.wikipedia.org/wiki/High_availability) system because it doesn't use a consensus protocol like Raft or Paxos to enforce strong consensus and consistency guarantees. However, even within the category of H.A systems, there is a broad spectrum of consistency and durability guarantees that a database can provide. Like many other H.A systems, M3DB allows database operators to tune the consistency and durability guarantees of the system in order to achieve the desired balance between performance, availability, durability, and consistency.
+
+Generally speaking, all of the default and example YAML configuration files for M3DB are configured to favor performance and availability over consistency and durability since that is usually the tradeoff that makes most sense for metrics and observability workloads. Database operators who are using M3DB for workloads that require stricter consistency and durability guarantees should consider tuning the default configuration to better suite their use case. The rest of this document describes the various configuration options that are available to M3DB operators to make those tradeoffs.
+
+We recommend referring to [this sample M3 YAML configuration file](https://github.com/m3db/m3/blob/master/src/dbnode/config/m3dbnode-all-config.yml) which has every possible configuration value set while reading the rest of this document to see how the described configurations fit into the YAML file as a whole.
+
+### Client Write and Read Consistency
+
+The possible configuration values for write and read consistency are discussed in more detail [in this section](../m3db/architecture/consistencylevels.md) of the documentation, but in short M3DB behaves similarly to other H.A systems with configurable consistency such as Cassandra that allow the caller to control the consistency level of writes and reads from the client.
+
+The most important thing to understand is that if you want the guarantee that you will be able to read the outcome of every succesful write then you must perform both writes and read with a consistency level of `majority`.
+
+### Commitlog Configuration
+
+By default M3DB runs with an asynchronous commitlog such that writes will be acknowleged as successful by the client even though the data may not have been physically flushed to the commitlog on disk yet. M3DB supports changing this default behavior to run the commitlog synchronously, but this is not currently exposed to users in the YAML configuration and generally leads to a massive performance degradation.
+
+M3DB operators can control when M3DB will attempt to flush the commitlog via the `commitlog` section of the YAML configuration. For example the default configuration:
+
+```
+commitlog:
+	flushMaxBytes: 524288
```

Comment:
nit: change this YAML indentation to 2 or 4 spaces (the default config itself is inconsistent)